crates/fluvio-auth/src/x509/authenticator.rs

```diff
@@ -90,7 +90,7 @@ impl X509Authenticator {
         let principal = Self::principal_from_raw_certificate(
             &client_certificate
                 .to_der()
-                .map_err(|err| err.into_io_error())?,
+                .map_err(|err| IoError::new(IoErrorKind::Other, err))?,
```

Comment:
can this be changed to anyhow?

Reply:
Yes, but will eventually need to be wrapped in IoError. I think some major changes would be required in `fluvio-socket` to avoid returning `std::io::Error` from the `authenticate` function, which would probably be best reviewed in a separate PR

Reply:
sound good